softvis3d-frontend/src/components/scene/Scene.tsx

```diff
@@ -36,82 +36,102 @@ export default class Scene extends React.Component<SceneProps, SceneStates> {
                 <canvas id={SoftVis3dScene.CANVAS_ID}
-                        onMouseDown={() => { this.mouseMoved = false; }}
-                        onMouseMove={() => { this.mouseMoved = true; }}
-                        onMouseUp={(e) => { this.onMouseUp(e); }}
-                        />
+                        onMouseDown={() => { this._mouseActions.setMouseMoved(false); }}
+                        onMouseMove={() => { this._mouseActions.setMouseMoved(true); }}
+                        onMouseUp={(e) => { this._mouseActions.onMouseUp(e); }}
+                />
                 <SceneInformation sceneStore={sceneStore}/>
-           </div>
+            </div>
         );
     }
 
-    // public for tests
     public updateCameraPosition() {
-        this.props.sceneStore.cameraPosition = this.props.sceneStore.scenePainter.getCamera().position;
+        this.props.sceneStore.cameraPosition = this._threeSceneService.getCameraPosition();
     }
 
-    public handleMouseDown(event: MouseEvent) {
-        const self = document.getElementById("scene-container");
-        let isWithinScene = event.target === self || HtmlDom.isDescendant(self, event.target as HTMLElement);
+    public handleMouseDown(event: Event<boolean>) {
+        let isWithinScene: boolean = event.getType();
         if (isWithinScene ? !this.state.focus : this.state.focus) {
             this.setState({...this.state, focus: isWithinScene});
         }
     }
 
-    public handleKeyDown(event: KeyboardEvent) {
+    /**
+     * Test injection setter
+     */
+
+    public set threeSceneService(value: ThreeSceneService) {
+        this._threeSceneService = value;
+    }
+
+    public set mouseActions(value: SceneMouseInteractions) {
+        this._mouseActions = value;
+    }
+
+    public set keyActions(value: SceneKeyInteractions) {
+        this._keyActions = value;
+    }
+
+    /**
+     * private methods
+     */
+
+    private selectObject(event: Event<MouseEvent>) {
+        this.props.sceneStore.selectedObjectId = this._threeSceneService.makeSelection(event.getType());
+    }
+
+    private resetCamera() {
         if (!this.state.focus) {
             return;
         }
 
-        switch (event.keyCode) {
-            case Scene.KEY_CODE_R:
-                this.props.sceneStore.scenePainter.resetCameraPosition(this.props.sceneStore.shapes);
-                break;
-            case Scene.KEY_CODE_L:
-                this.setState({...this.state, legend: !this.state.legend});
-                break;
-            default:
-                // KEY NOT REGISTERED
-        }
+        this._threeSceneService.resetCameraPosition(this.props.sceneStore.shapes);
     }
 
-    private onMouseUp(event: any) {
-        this.updateCameraPosition();
-
-        if (!this.mouseMoved) {
-            this.props.sceneStore.selectedObjectId = this.props.sceneStore.scenePainter.makeSelection(event);
+    private toggleLegend() {
+        if (!this.state.focus) {
```

Comment:
The focus state should not be duplicated within the methods `resetCamera` and `toggleLegend` as it does not interfere with the _pure_ action. I think it should be part of the event propagation process.